### PR TITLE
Change mod_fluka error handling, fix DA version

### PR DESCRIPTION
**Pull request by JamesMolson**

### CODE REVIEW

File: source/mod_fluka.f90 [hunk 1]
```diff
@@ -1099,9 +1099,9 @@ subroutine kernel_fluka_element( nturn, i, ix )
            napx, xv1, xv2, yv1, yv2, sigmv, ejv, naa, nzz, nucm, nqq, pdgid, &
            spin_x, spin_y, spin_z )
 
-      if (ret.eq.-1) then
-         write(lerr,'(A)')'FLUKA> ERROR -1 in Fluka communication returned by fluka_send_receive...'
-         write(fluka_log_unit,'(A)')'# Error -1 in Fluka communication returned by fluka_send_receive...'
+      if (ret.lt.0) then
+         write(lerr,'(A,i4,A)')'FLUKA> ERROR ', ret, ' in Fluka communication returned by fluka_send_receive...'
```
Comment (vkbo) [inline source/mod_fluka.f90]: Why not use format `i0` so there is no chance of `****`?

File: source/sixda.f90 [hunk 2]
```diff
@@ -720,6 +720,7 @@ subroutine runda
 #include "include/dalin3.f90"
 #include "include/dalin4.f90"
 #include "include/dalin5.f90"
+          else
```
Comment (amereghe) [inline source/sixda.f90]: "Thanks, God, it's Christmas!" (the Queen)

File: source/mod_fluka.f90 (hunk 1 above)
Comment (amereghe) [reply source/mod_fluka.f90]: I agree with you. @vkbo - even though I don't think we will have many error codes, but it is better to set it once correctly forever.

File: source/mod_fluka.f90 [hunk 3]
```diff
@@ -304,7 +304,7 @@ subroutine fluka_end()
     ! Send end of computation
     n = ntsendeoc(fluka_cid)
     if(n.lt.0) then
-      write(fluka_log_unit,*) "# FlukaIO error: Error sending End of Computation"
+      write(fluka_log_unit,*) "# FlukaIO error: ", n, " - Error sending End of Computation"
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

File: source/mod_fluka.f90 [hunk 4]
```diff
@@ -313,8 +313,8 @@ subroutine fluka_end()
     n = ntwait(fluka_cid, mtype, &
           flid, flgen, flwgt, flx, fly, flz, flxp, flyp, &
           flm, flpc, flt, flpdgid, flq, flsx, flsy, flsz)
-    if(n.eq.-1) then
-      write(fluka_log_unit,*) "# FlukaIO error: Server timed out while waiting End of Computation"
+    if(n.lt.0) then
+      write(fluka_log_unit,*) "# FlukaIO error: ", n, " - Server timed out while waiting End of Computation"
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

File: source/mod_fluka.f90 [hunk 5]
```diff
@@ -411,8 +411,8 @@ integer function fluka_send(turn, ipt, el, npart, xv1, xv2, yv1, yv2, s, etot, a
     fluka_last_rcvd_mess = -1
 
     n = ntsendipt(fluka_cid, turn, ipt)
-    if(n.eq.-1) then
-      write(fluka_log_unit,*) "# FlukaIO error: Error sending Insertion Point"
+    if(n.lt.0) then
+      write(fluka_log_unit,*) "# FlukaIO error: ", n, " - Error sending Insertion Point"
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

File: source/mod_fluka.f90 [hunk 6]
```diff
@@ -481,8 +481,8 @@ integer function fluka_send(turn, ipt, el, npart, xv1, xv2, yv1, yv2, s, etot, a
             flm, flet, flt, &
             flpdgid, flq, flsx, flsy, flsz)
 
-      if(n.eq.-1) then
-        write(fluka_log_unit,*) "# FlukaIO error: Error sending Particle"
+      if(n.lt.0) then
+        write(fluka_log_unit,*) "# FlukaIO error: ", n, " - Error sending Particle"
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

File: source/mod_fluka.f90 [hunk 7]
```diff
@@ -497,7 +497,7 @@ integer function fluka_send(turn, ipt, el, npart, xv1, xv2, yv1, yv2, s, etot, a
     n = ntsendeob(fluka_cid)
 
     if(n.lt.0) then
-      write(fluka_log_unit,*) "# FlukaIO error: Error sending End of Batch"
+      write(fluka_log_unit,*) "# FlukaIO error: ", n, " - Error sending End of Batch"
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

File: source/mod_fluka.f90 [hunk 8]
```diff
@@ -590,8 +590,8 @@ integer function fluka_receive(turn, ipt, el, napx, xv1, xv2, yv1, yv2, s, etot,
               flm, flet, flt, &
               flpdgid, flq, flsx, flsy, flsz)
 
-      if(n.eq.-1) then
-        write(fluka_log_unit,*) "# FlukaIO error: Server timed out while waiting for message"
+      if(n.lt.0) then
+        write(fluka_log_unit,*) "# FlukaIO error: ", n ," - Server timed out while waiting for message"
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

File: source/mod_fluka.f90 [hunk 9]
```diff
@@ -1099,9 +1099,9 @@ subroutine kernel_fluka_element( nturn, i, ix )
            napx, xv1, xv2, yv1, yv2, sigmv, ejv, naa, nzz, nucm, nqq, pdgid, &
            spin_x, spin_y, spin_z )
 
-      if (ret.eq.-1) then
-         write(lerr,'(A)')'FLUKA> ERROR -1 in Fluka communication returned by fluka_send_receive...'
-         write(fluka_log_unit,'(A)')'# Error -1 in Fluka communication returned by fluka_send_receive...'
+      if (ret.lt.0) then
+         write(lerr,'(A,i4,A)')'FLUKA> ERROR ', ret, ' in Fluka communication returned by fluka_send_receive...'
+         write(fluka_log_unit,'(A,i4,A)')'# Error ', ret, ' in Fluka communication returned by fluka_send_receive...'
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

File: source/mod_fluka.f90 [hunk 10]
```diff
@@ -1262,9 +1262,9 @@ subroutine kernel_fluka_entrance( nturn, i, ix )
            napx, xv1, xv2, yv1, yv2, sigmv, ejv, naa, nzz, nucm, nqq, pdgid, &
            spin_x, spin_y, spin_z )
 
-      if (ret.eq.-1) then
-         write(lerr,'(A)')'FLUKA> ERROR -1 in Fluka communication returned by fluka_send...'
-         write(fluka_log_unit,'(A)')'# Error -1 in Fluka communication returned by fluka_send...'
+      if (ret.lt.0) then
+         write(lerr,'(A,i4,A)')'FLUKA> ERROR ', ret,' in Fluka communication returned by fluka_send...'
+         write(fluka_log_unit,'(A,i4,A)')'# Error ', ret, ' in Fluka communication returned by fluka_send...'
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

File: source/mod_fluka.f90 [hunk 11]
```diff
@@ -1317,9 +1317,9 @@ subroutine kernel_fluka_exit
            napx, xv1, xv2, yv1, yv2, sigmv, ejv, naa, nzz, nucm, nqq, pdgid, &
            spin_x, spin_y, spin_z )
 
-      if (ret.eq.-1) then
-         write(lerr,'(A)')'FLUKA> ERROR -1 in Fluka communication returned by fluka_receive...'
-         write(fluka_log_unit,'(A)')'# Error -1 in Fluka communication returned by fluka_receive...'
+      if (ret.lt.0) then
+         write(lerr,'(A,i4,A)')'FLUKA> ERROR ', ret, ' in Fluka communication returned by fluka_receive...'
+         write(fluka_log_unit,'(A,i4,A)')'# Error ',ret, ' in Fluka communication returned by fluka_receive...'
```
Comment (amereghe) [inline source/mod_fluka.f90]: `write(fluka_log_unit,'(A,i0,A)')`

**Pull request merged**